src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -4164,7 +4326,13 @@ UpdateConnectionWaitFlags(WorkerSession *session, int waitFlags)
 		return;
 	}
 
+#if PG_VERSION_NUM >= PG_VERSION_15
+
+	/* always detect closed sockets */
+	connection->waitFlags = waitFlags | WL_SOCKET_CLOSED;
```

Comment:
Should the changes to this function be backported? i.e. by not waiting for WL_SOCKET_CLOSED do we miss failures?

Reply:
I think we'd want these changes to go through a release testing given the points it touches